script/draft-release/run.ts

```diff
@@ -0,0 +1,108 @@
+import { sort as semverSort, SemVer } from 'semver'
+
+import { spawn } from '../changelog/spawn'
+import { getLogLines } from '../changelog/git'
+import {
+  convertToChangelogFormat,
+  getChangelogEntriesSince,
+} from '../changelog/parser'
+
+import { Channel } from './channel'
+import { getNextVersionNumber } from './version'
+
+const jsonStringify: (obj: any) => string = require('json-pretty')
+
+async function getLatestRelease(excludeBetaReleases: boolean): Promise<string> {
+  const allTags = await spawn('git', ['tag'])
+  let releaseTags = allTags
+    .split('\n')
+    .filter(tag => tag.startsWith('release-'))
+    .filter(tag => tag.indexOf('-linux') === -1)
+    .filter(tag => tag.indexOf('-test') === -1)
+
+  if (excludeBetaReleases) {
+    releaseTags = releaseTags.filter(tag => tag.indexOf('-beta') === -1)
+  }
+
+  const releaseVersions = releaseTags.map(tag => tag.substr(8))
+
+  const sortedTags = semverSort(releaseVersions)
+  const latestTag = sortedTags[sortedTags.length - 1]
+
+  if (latestTag instanceof SemVer) {
+    return latestTag.raw
+  }
+
+  return latestTag
+}
+
+function parseChannel(arg: string): Channel {
+  if (arg === 'production' || arg === 'beta') {
+    return arg
+  }
+
+  throw new Error(`An invalid channel ${arg} has been provided`)
+}
+
+function printInstructions(nextVersion: string, entries: Array<string>) {
```

Comment:
This function can be made a bit clearer:

```ts
function printInstructions(nextVersion: string, entries: Array<string>) {
  const obj: any = {}
  obj[`${nextVersion}`] = entries.sort()

  const steps = [
    `Ensure that the value of the 'version' property in app/package.json is '${nextVersion}'`,
    `Add the following entry to changelog.json as a starting point`,
    `${jsonStringify(obj)}`,
    'Modify the release notes so that they make sense and contain only user-facing changes',
    `Commit the changes and push them to GitHub'`,
    `Read this to perform the release: https://github.com/desktop/desktop/blob/master/docs/process/releasing-updates.md`,
  ]
    .map((value, index) => `${index + 1}. ${value}`)
    .join('\n')

  console.log(steps)
}
```

Reply:
While this is a bit terse, it will render `${jsonStringify(obj)}` with it's own number, which we want to avoid.

I'll have a go at tidying this up a bit anyway so it's easier to follow.

Reply:
13bd78c5c

Reply:
This may seem silly/hacky, but this could work...
`${jsonStringify(obj)` -> `#${jsonStringify(obj)}`
Then replace anything that matches the regex `/\d.\s#/` with the `empty string`